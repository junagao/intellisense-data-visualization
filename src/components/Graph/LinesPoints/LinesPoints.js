import React from 'react';
import PropTypes from 'prop-types';
import { v4 as uuidv4 } from 'uuid';

const LinesPoints = ({
  metric,
  dataset,
  marginLeft,
  color,
  scales,
  onHover,
}) => {
  const { xScale, yScale } = scales;
  return (
    <g fill={color}>
      {dataset.map(({ time, value }) => (
        <circle
          key={uuidv4()}
          cx={xScale(time) - marginLeft}
          cy={yScale(value)}
          r="3"
          stroke="#fff"
          onMouseOver={() => onHover(metric, color, time, value)}
          onFocus={() => onHover(metric, color, time, value)}
          onMouseOut={() => onHover(metric, color, time, value)}
          onBlur={() => onHover(metric, color, time, value)}
        />
      ))}
    </g>
  );
};

LinesPoints.propTypes = {
  color: PropTypes.string.isRequired,
  dataset: PropTypes.arrayOf(
    PropTypes.shape({
      time: PropTypes.number,
      value: PropTypes.number,
    }),
  ).isRequired,
  marginLeft: PropTypes.number.isRequired,
  metric: PropTypes.string.isRequired,
  onHover: PropTypes.func.isRequired,
  scales: PropTypes.shape({
    xScale: PropTypes.func,
    yScale: PropTypes.func,
  }).isRequired,
};

export default LinesPoints;
