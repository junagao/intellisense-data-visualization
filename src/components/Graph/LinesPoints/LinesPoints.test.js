import React from 'react';
import { shallow } from 'enzyme';

import LinesPoints from './LinesPoints';

describe('LinesPoints', () => {
  let wrapper;

  const mockMetric = 'metric A';
  const mockDataset = [
    { time: 0, value: 23 },
    { time: 10, value: 53 },
  ];
  const mockColor = 'red';
  const mockMarginLeft = 30;
  const xScale = jest.fn();
  const yScale = jest.fn();
  const mockOnHover = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <LinesPoints
        metric={mockMetric}
        dataset={mockDataset}
        marginLeft={mockMarginLeft}
        color={mockColor}
        scales={{ xScale, yScale }}
        onHover={mockOnHover}
      />,
    );
  });

  it('should render correctly given the required props', () => {
    expect(wrapper).toMatchSnapshot();
  });

  it('should render one g element', () => {
    expect(wrapper.find('g').length).toEqual(1);
  });

  it('should render two circle elements', () => {
    expect(wrapper.find('circle').length).toEqual(2);
  });
});
