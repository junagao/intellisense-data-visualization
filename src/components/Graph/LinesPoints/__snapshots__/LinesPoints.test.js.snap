// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LinesPoints should render correctly given the required props 1`] = `
<g
  fill="red"
>
  <circle
    cx={NaN}
    key="0"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    r="3"
    stroke="#fff"
  />
  <circle
    cx={NaN}
    key="10"
    onBlur={[Function]}
    onFocus={[Function]}
    onMouseOut={[Function]}
    onMouseOver={[Function]}
    r="3"
    stroke="#fff"
  />
</g>
`;
